v2/pkgs/tools.go

```diff
@@ -284,44 +285,24 @@ func rename(base string) extract.Renamer {
 	}
 }
 
-func readInstalled(installedFile string) (map[string]string, error) {
+func (t *Tools) readInstalled() error {
+	t.mutex.Lock()
```

Comment:
Shouldn't we use [`RLock()`](https://pkg.go.dev/sync#RWMutex.RLock) here, since we are reading?

Reply:
Fixed